mono/sgen/sgen-workers.c

```diff
@@ -210,6 +210,37 @@ workers_get_work (WorkerData *data)
 	return FALSE;
 }
 
+static gboolean
+workers_steal_work (WorkerData *data)
+{
+	SgenMajorCollector *major;
+
+	g_assert (sgen_gray_object_queue_is_empty (&data->private_gray_queue));
+
+	/* If we're parallel, steal from other workers' private gray queues  */
+	major = sgen_get_major_collector ();
+	if (major->is_parallel) {
```

Comment:
Exit early if possible, to avoid nesting.  As in

    if (!major->is_parallel)
        return FALSE;